src/natcap/invest/utils.py

```diff
@@ -795,3 +795,65 @@ def _assert_vectors_equal(
         expected_vector = None
 
     return None
+
+def _reclassify_raster_op(
+        raster_path_band, value_map, target_raster_path, target_datatype,
+        target_nodata, error_details=None, values_required=True):
+    """A wrapper function for calling ``pygeoprocessing.reclassify_raster``.
+
+    This wrapper function is helpful when added as a ``TaskGraph.task`` so
+    a better error message can be provided to the users if an exception
+    occurs.
+
+    Args:
+        raster_path_band (tuple): a tuple including file path to a raster
+            and the band index to operate over. ex: (path, band_index)
+        value_map (dictionary): a dictionary of values of
+            {source_value: dest_value, ...} where source_value's type is the
+            same as the values in ``base_raster_path`` at band ``band_index``.
+            Must contain at least one value.
+        target_raster_path (string): target raster output path; overwritten if
+            it exists
+        target_datatype (gdal type): the numerical type for the target raster
+        target_nodata (numerical type): the nodata value for the target raster
+            Must be the same type as target_datatype
+        error_details (dict): a dictionary with key value pairs that provide 
+            more context for a raised
+            ``pygeoprocessing.ReclassificationMissingValuesError``.
+            keys must be {'raster_name', 'column_name', 'table_name'}. Values 
+            each key represent:
+                'raster_name' - string for the raster name being reclassified
+                'column_name' - column name from table with which ``value_map``
+                    keys came from. 
+                'table_name' - table name that ``value_map`` came from.
+        values_required (bool): If True, raise a ValueError if there is a
+            value in the raster that is not found in ``value_map``.
+
+    Returns:
+        None
+
+    Raises:
+        ValueError if ``values_required`` is ``True`` and a pixel value from
+        ``raster_path_band`` is not a key in ``value_map``.
+    """
+    try:
+        pygeoprocessing.reclassify_raster(
+            raster_path_band, value_map, target_raster_path, target_datatype,
+            target_nodata, values_required=values_required)
+    except pygeoprocessing.ReclassificationMissingValuesError as err:
+        if error_details is None:
+            raster_name = ' '
+            column_name = ' '
+            table_name = ' '
+        else:
+            raster_name = ' ' + error_details.get('raster_name') + ' '
+            column_name = " '" + error_details.get('column_name') + "' " 
+            table_name = ' ' + error_details.get('table_name') + ' '  
```

Comment:
Oh geez, I had a default empty string value in there for the `get` call but I must have removed it when I was trying to better string format the messages. Thanks for pointing this out. I'll take a look.

Reply:
Hey @phargogh I updated the handling of the dictionary argument here: https://github.com/natcap/invest/pull/287/commits/06f3488ef9b3f9df2c63f22a21c1c4c6858506d8. 

It doesn't tackle your larger question of: 
>Is it reasonable behavior to not provide this error information?

I guess I've set it up to default to a generic InVEST reclassification error message that is hopefully more helpful then what was previously returned. But yea, this is a fair point. I guess if it was required and you didn't want to specify labels you could pass in a dictionary with empty string values... I don't have very strong feelings about either way.

Reply:
@dcdenu4 @phargogh , here's my thoughts on this convo. I think it's reasonable to require all the error details data. After all, that seems like the main reason I would want to use this wrapper function instead of the original pygeoprocessing one. If you let me be lazy and not fill out the details, well I might do just that :) 

Also, I think it's important that when we catch an exception (i.e. pygeoprocessing's `ReclassificationMissingValuesError`) to re-raise it, we're always adding _more_ useful context and never hiding anything useful that we would have seen otherwise. For example, the original exception message -- while written for programmers rather than invest users -- does include the input raster path, but I think the re-raised exception without any `error_details` might hide that.

> also if all of these fields are required, then maybe it would make sense to have them be required parameters in the function?

:+1:

Reply:
Ah, those are both really good points @davemfish !  I agree, and also with the idea about having them be required, separate parameters in the function.